### PR TITLE
Pass MSVS environment variables through to make system

### PR DESCRIPTION
**Pull request by sxa555**
Forward port of https://github.com/ibmruntimes/openj9-openjdk-jdk9/commit/f9aeff3d3179aa61e67dbde5413bcb6ef1c69665#diff-d6f51dbe595d728e2d1111f036170369 since I think this is what will resolve https://github.com/AdoptOpenJDK/openjdk-build/issues/281
FYI @vsebe

### CODE REVIEW

**Comment by sxa555** [comment]
I think that was because the initial attempt was based off a mirror of the master branch instead of the openj9 branch in the ibmruntimes extensions repo, although I've got another error now. Clearing the workspace and retrying ...

**Comment by jdekonin** [comment]
$(EXPORT_MSVS_BUILD_ENV_VARS) should be $(EXPORT_MSVS_ENV_VARS)

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -365,7 +365,7 @@ run-preprocessors-j9 : stage-j9 \
 		$(OUTPUTDIR)/vm/compiler/jit.version \
 		$(OUTPUTDIR)/vm/include/openj9_version_info.h
 	@$(ECHO) Running OpenJ9 preprocessors with OPENJ9_BUILDSPEC: $(OPENJ9_BUILDSPEC)
-	(export BOOT_JDK=$(BOOT_JDK) \
+	(export BOOT_JDK=$(BOOT_JDK) $(EXPORT_MSVS_BUILD_ENV_VARS) \
```
Comment (jdekonin) [inline closed/OpenJ9.gmk]: should be $(EXPORT_MSVS_ENV_VARS)

File: closed/OpenJ9.gmk [hunk 2]
```diff
@@ -382,7 +382,7 @@ run-preprocessors-j9 : stage-j9 \
 
 build-j9 : run-preprocessors-j9
 	@$(ECHO) Compiling OpenJ9 in $(OUTPUTDIR)/vm
-	(export OPENJ9_BUILD=true $(EXPORT_NO_USE_MINGW) \
+	(export OPENJ9_BUILD=true $(EXPORT_NO_USE_MINGW) $(EXPORT_MSVS_BUILD_ENV_VARS) \
```
Comment (jdekonin) [inline closed/OpenJ9.gmk]: should be $(EXPORT_MSVS_ENV_VARS)

**Comment by sxa555** [comment]
Thanks - fixed :-)

**Comment by sxa555** [comment]
Maybe that's one for @DanHeidinga or @andrew-m-leonard to answer

File: closed/autoconf/custom-hook.m4 [hunk 3]
```diff
@@ -17,8 +17,15 @@
 
 AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
 [
-  JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+ if test "x$OPENJDK_BUILD_OS_ENV" = xwindows.cygwin; then
   OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
```
Comment (vsebe) [inline closed/autoconf/custom-hook.m4]: OPENJDK_BUILD_JAVA_BASE_LDFLAGS is defined twice. Delete this line!

**Comment by jdekonin** [comment]
This is the historical reference to the other commit that was missing in jdk10.
https://github.com/ibmruntimes/openj9-openjdk-jdk9/pull/64

**Pull request merged**